mesh/mesh.go

```diff
@@ -352,13 +351,14 @@ func (m *Mesh) handleOrphanBlocks(blk *types.BlockHeader) {
 	}
 	m.orphanBlocks[blk.Layer()][blk.ID()] = struct{}{}
 	m.Info("Added block %d to orphans", blk.ID())
-	atomic.AddInt32(&m.orphanBlockCount, 1)
 	for _, b := range blk.ViewEdges {
-		for _, layermap := range m.orphanBlocks {
+		for layerId, layermap := range m.orphanBlocks {
 			if _, has := layermap[b]; has {
 				m.Log.Debug("delete block ", b, "from orphans")
 				delete(layermap, b)
-				atomic.AddInt32(&m.orphanBlockCount, -1)
+				if len(layermap) == 0{
+					delete(m.orphanBlocks, layerId)
```

Comment:
Is it ok to modify the map while you iterate it? Usually, it is a bad practice, though I'm not sure how it is considered in Go

Reply:
i think that in this case it wont cause an issue but it is bad practice, 
IMO better to use a for loop on the keys and access the map with key instead of range

Reply:
its OK to delete a member of a map while iterating in go, iterating over keys will not change the fact that were deleting items while iterating them

Reply:
like i said it works, but its not the best practice.
there are caveats with mutating maps while using range just need to be aware 
for instance 
"If a map entry is created during iteration, that entry may be produced during the iteration or may be skipped. The choice may vary for each entry created and from one iteration to the next"

Reply:
from effective go:
https://golang.org/doc/effective_go.html#for

Reply:
im aware, still IMO more prone to errors while maintaining the code